services/repository/branch.go

```diff
@@ -130,10 +163,17 @@ func loadOneBranch(ctx context.Context, repo *repo_model.Repository, dbBranch *g
 
 	// it's not default branch
 	if repo.DefaultBranch != dbBranch.Name && !dbBranch.IsDeleted {
-		var err error
-		divergence, err = files_service.CountDivergingCommits(ctx, repo, git.BranchPrefix+branchName)
-		if err != nil {
-			log.Error("CountDivergingCommits: %v", err)
+		divergence = getDivergenceFromCache(repo.ID, dbBranch.Name)
+		if divergence.Ahead == -1 && divergence.Behind == -1 {
```

Comment:
Not block, but I don't think it's a good idea to use a magic value to indicate a cache miss.

Maybe a nil return or a bool flag.

Reply:
9060703128dc8c368d3b66d7ad3476f8db48267e